osu.Game/Screens/SelectV2/Panel.cs

```diff
@@ -186,27 +187,20 @@ public PulsatingBox()
                 };
             }
 
-            private int separation = 1;
-
             protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, ChannelAmplitudes amplitudes)
             {
                 base.OnNewBeat(beatIndex, timingPoint, effectPoint, amplitudes);
 
-                if (beatIndex % separation != 0)
+                if (beatIndex % Math.Pow(2, FlashOffset) != 0)
```

Comment:
I'm not sure how well this concept works for anything that's not in common time (think waltz time or something even wackier than that like a 7/4) but I guess users will be the judge of that?